pkg/restapi/v1/devapi/controller.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+//go:generate oapi-codegen --config=openapi.cfg.yaml ../../../../docs/v1/openapi.yaml
+//go:generate mockgen -destination controller_mocks_test.go -self_package mocks -package devapi -source=controller.go -mock_names didConfigService=MockDidConfigService
+
+package devapi
+
+import (
+	"context"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+
+	apiUtil "github.com/trustbloc/vcs/pkg/restapi/v1/util"
+	"github.com/trustbloc/vcs/pkg/service/didconfiguration"
+)
+
+type didConfigService interface {
+	DidConfig(
+		ctx context.Context,
+		profileType didconfiguration.ProfileType,
+		profileID string,
+		contextURL string,
+	) (*didconfiguration.DidConfiguration, error)
+}
+
+type Config struct {
+	DidConfigService didConfigService
+}
+
+type Controller struct {
+	didConfigService didConfigService
+}
+
+func NewController(
+	config *Config,
+) *Controller {
+	return &Controller{
+		didConfigService: config.DidConfigService,
+	}
+}
+
+// DidConfig requests well-known DID config.
+// GET /{profileType}/profiles/{profileID}/well-known/did-config.
+func (c *Controller) DidConfig(ctx echo.Context, profileType string, profileID string) error {
+	var contextURL strings.Builder
+
+	if scheme := ctx.Request().URL.Scheme; len(scheme) > 0 {
+		contextURL.WriteString(scheme + "://")
+	} else {
+		contextURL.WriteString("https://")
+	}
+
+	if hostName := ctx.Request().URL.Hostname(); len(hostName) > 0 {
+		contextURL.WriteString(hostName + "/")
+	} else {
+		contextURL.WriteString("localhost")
```

Comment:
this looks strange in production code, plz elaborate on this logic

Reply:
I agree with that

Reply:
We need to build absolute URI. in HTTP request, standard Request().URL there is no possibility of receiving full URL (RequestURI() in some cases can return only Path, without schema and domain, so I combine URL by myself from request, that code basically can be moved to some helper. cc @aholovko @fqutishat

Reply:
Stas, you are using that URL as a value of @context - I think it's not correct, there should be constant value "https://identity.foundation/.well-known/did-configuration/v1" (JSON-LD context for "DID Configuration" document). Probably, you confused that with https://identity.foundation/.well-known/resources/did-configuration/#did-configuration-uri that describes where that document should be hosted. cc @skynet2

Reply:
you are right @aholovko , fixed